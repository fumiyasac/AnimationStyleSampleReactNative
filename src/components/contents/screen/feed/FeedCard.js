/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import React, { Component } from 'react';

import {
  StyleSheet,
  View,
  Text
} from 'react-native';

import { 
  Card
} from 'native-base';

export default class FeedCars extends React.Component {
  render() {
    return (
      <View>
        <Card>
          <Text>{"テスト"}</Text>
        </Card>
      </View>
    );
  }
}

const styles = StyleSheet.create({});
