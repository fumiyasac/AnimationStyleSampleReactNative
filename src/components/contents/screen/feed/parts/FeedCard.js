/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import React, { Component } from 'react';

import {
  StyleSheet,
  Image,
  View,
  TouchableOpacity,
  Dimensions
} from 'react-native';

import { 
  Content,
  Card,
  CardItem,
  Left,
  Body,
  Button,
  Right,
  Text,
  Icon,
  Thumbnail
} from 'native-base';

const {
  width: DEVICE_WIDTH
} = Dimensions.get('window');

const imageAreaHeight = DEVICE_WIDTH * 0.6;

export default class FeedCard extends React.Component {
  render() {
    const { feed, onPressLikeButton, onPressCommentButton, onPressReadModeButton } = this.props;
    const { title, subtitle, category, detail, image_url, created } = feed;
    return (
      <Content>
        <Card>
          {/* Header */}
          <CardItem>
            <Body>
              <Text style={styles.title}>{title}</Text>
              <Text style={styles.subtitle} note>{subtitle}</Text>
              <Text style={styles.category} note>{category}</Text>
            </Body>
          </CardItem>
          {/* Thumbnail */}
          <CardItem style={styles.cardImageWrapper} cardBody>
            <Image style={styles.cardImage} source={ { uri: image_url } } />
          </CardItem>
          {/* Detail */}
          <CardItem>
            <Text style={styles.detail}>{detail}</Text>
          </CardItem>
          {/* Footer */}
          <CardItem style={styles.cardFooter}>
            <Left>
              <Text style={styles.dayText}>{created}</Text>
            </Left>
            <Body>
              <View style={styles.centerBase}>
                <Button transparent onPress={onPressLikeButton}>
                  <View style={styles.buttonBase}>
                    <Icon style={styles.heartButtonIcon} name="heart" />
                    <Text style={styles.heartButtonText}>123</Text>
                  </View>
                </Button>
                <Button transparent onPress={onPressCommentButton}>
                  <View style={styles.buttonBase}>
                    <Icon style={styles.commentButtonIcon} name="chatbubbles" />
                    <Text style={styles.commentButtonText}>456</Text>
                  </View>
                </Button>
              </View>
            </Body>
            <Right>
              <View style={styles.rightBase}>
                <TouchableOpacity onPress={onPressReadModeButton}>
                  <Text style={styles.readmoreText}>続きを読む</Text>
                </TouchableOpacity>
              </View>
            </Right>
          </CardItem>
        </Card>
      </Content>
    );
  };
}

const styles = StyleSheet.create({
  title: {
    marginTop: 3,
    fontSize: 15,
    fontWeight: "bold"
  },
  subtitle: {
    marginTop: 4,
    marginTop: 8,
    fontSize: 12
  },
  category: {
    marginTop: 8,
    marginBottom: 4,
    fontSize: 12
  },
  cardImageWrapper: {
    marginLeft: 12,
    marginRight: 12,   
  },
  cardImage: {
    width: null,
    height: imageAreaHeight,
    flex: 1
  },
  detail: {
    marginTop: 3,
    fontSize: 13,
    lineHeight: 20
  },
  cardFooter: {
    marginTop: -16
  },
  centerBase: {
    flexDirection: 'row'
  },
  dayText: {
    color: '#999999',
    marginLeft: 0,
    fontSize: 13
  },
  buttonBase: {
    flexDirection: 'row'
  },
  heartButtonIcon: {
    color: '#f08080'
  },
  heartButtonText: {
    fontSize: 13,
    color: '#f08080',
    marginTop: 2,
    marginLeft: -10
  },
  commentButtonIcon: {
    color: '#ffc125'
  },
  commentButtonText: {
    fontSize: 13,
    color: '#ffc125',
    marginTop: 2,
    marginLeft: -10
  },
  rightBase: {
    flexDirection: 'row'
  },
  readmoreText: {
    textAlign: 'right',
    color: '#999999',
    marginLeft: 0,
    fontSize: 13,
    textDecorationLine: 'underline'
  }
});
