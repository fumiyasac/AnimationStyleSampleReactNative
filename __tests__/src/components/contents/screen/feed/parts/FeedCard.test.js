import 'react-native';
import React from 'react';
import FeedCard from '../../../../../../../src/components/contents/screen/feed/parts/FeedCard';

import renderer from 'react-test-renderer';

it('FeedCardコンポーネントが正しく表示されること', () => {

  let stubObject = { 
    "feed": {
      "contents": [
        {
          "id": "1",
          "title": "タイトルが入ります。",
          "subtitle": "サブタイトルが入ります。",
          "category": "カテゴリーが入ります。",
          "detail": "詳細テキストが入ります。詳細テキストが入ります。詳細テキストが入ります。詳細テキストが入ります。詳細テキストが入ります。詳細テキストが入ります。詳細テキストが入ります。詳細テキストが入ります。詳細テキストが入ります。詳細テキストが入ります。",
          "image_url" : "https://otsuka-shop.s3-ap-northeast-1.amazonaws.com/shops/images/1/large.jpg",
          "created": "YYYY/MM/DD"
        }
      ]
    }
  };

  const onPressLikeButtonMock = jest.fn();
  const onPressCommentButtonMock = jest.fn();
  const onPressReadModeButtonMock = jest.fn();

  const tree = renderer.create(
    <FeedCard feed={stubObject.feed.contents[0]}
      onPressLikeButton ={onPressLikeButtonMock()}
      onPressCommentButton ={onPressCommentButtonMock()}
      onPressReadModeButton ={onPressReadModeButtonMock()}
      />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
  