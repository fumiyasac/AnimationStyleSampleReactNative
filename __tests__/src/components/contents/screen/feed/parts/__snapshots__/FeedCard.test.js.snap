// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedCardコンポーネントが正しく表示されること 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Array [
      Object {
        "padding": undefined,
      },
      undefined,
    ]
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutomaticScroll={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  getScrollResponder={[Function]}
  handleOnScroll={[Function]}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  keyboardShouldPersistTaps="handled"
  keyboardSpace={0}
  onScroll={[Function]}
  resetKeyboardSpace={[Function]}
  resetScrollToCoords={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scrollEventThrottle={1}
  scrollForExtraHeightOnAndroid={[Function]}
  scrollToEnd={[Function]}
  scrollToFocusedInput={[Function]}
  scrollToPosition={[Function]}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
  viewIsInsideTabBar={false}
  virtual={undefined}
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 0.5,
          "borderColor": "#ccc",
          "borderLeftWidth": 0.5,
          "borderRadius": 2,
          "borderRightWidth": 0.5,
          "borderTopWidth": 0.5,
          "elevation": 3,
          "flex": 1,
          "flexWrap": "nowrap",
          "marginBottom": 5,
          "marginLeft": 2,
          "marginRight": 2,
          "marginTop": 5,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 1.5,
        }
      }
      virtual={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 2,
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
          }
        }
        virtual={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "stretch",
              "flex": 1,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                },
                Object {
                  "fontSize": 15,
                  "fontWeight": "bold",
                  "marginTop": 3,
                },
              ]
            }
            uppercase={false}
            virtual={undefined}
          >
            タイトルが入ります。
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            note={true}
            style={
              Array [
                Object {
                  "color": "#808080",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "200",
                  "marginRight": 20,
                },
                Object {
                  "fontSize": 12,
                  "marginTop": 8,
                },
              ]
            }
            uppercase={false}
            virtual={undefined}
          >
            サブタイトルが入ります。
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            note={true}
            style={
              Array [
                Object {
                  "color": "#808080",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "200",
                  "marginRight": 20,
                },
                Object {
                  "fontSize": 12,
                  "marginBottom": 4,
                  "marginTop": 8,
                },
              ]
            }
            uppercase={false}
            virtual={undefined}
          >
            カテゴリーが入ります。
          </Text>
        </View>
      </View>
      <View
        cardBody={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": -5,
              "paddingLeft": -5,
              "paddingRight": -5,
              "paddingTop": -5,
            },
            Object {
              "marginLeft": 12,
              "marginRight": 12,
            },
          ]
        }
        virtual={undefined}
      >
        <Image
          source={
            Object {
              "uri": "https://otsuka-shop.s3-ap-northeast-1.amazonaws.com/shops/images/1/large.jpg",
            }
          }
          style={
            Object {
              "flex": 1,
              "height": 450,
              "width": null,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 2,
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "fontSize": 13,
                "lineHeight": 20,
                "marginTop": 3,
              },
            ]
          }
          uppercase={false}
          virtual={undefined}
        >
          詳細テキストが入ります。詳細テキストが入ります。詳細テキストが入ります。詳細テキストが入ります。詳細テキストが入ります。詳細テキストが入ります。詳細テキストが入ります。詳細テキストが入ります。詳細テキストが入ります。詳細テキストが入ります。
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 2,
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
          }
        }
        virtual={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#f08080",
                    "fontSize": 16,
                  },
                ],
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
            virtual={undefined}
          >
            
          </Text>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "alignSelf": "center",
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "marginLeft": 10,
                  },
                  Object {
                    "color": "#f08080",
                    "fontSize": 13,
                    "marginLeft": 8,
                    "marginTop": -2,
                  },
                ]
              }
              uppercase={false}
              virtual={undefined}
            >
              いいね数
              : 
              123
              件
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 2,
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
          }
        }
        virtual={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#ffc125",
                    "fontSize": 16,
                  },
                ],
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
            virtual={undefined}
          >
            
          </Text>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "alignSelf": "center",
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "marginLeft": 10,
                  },
                  Object {
                    "color": "#ffc125",
                    "fontSize": 13,
                    "marginLeft": 8,
                    "marginTop": -2,
                  },
                ]
              }
              uppercase={false}
              virtual={undefined}
            >
              応援メッセージ数
              : 
              456
              件
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            },
            Object {
              "marginBottom": 12,
              "marginTop": 8,
            },
          ]
        }
        virtual={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "marginLeft": 10,
                },
                Object {
                  "color": "#999999",
                  "fontSize": 13,
                  "marginLeft": 0,
                },
              ]
            }
            uppercase={false}
            virtual={undefined}
          >
            2018.04.10
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "alignSelf": "center",
              "flex": 0.8,
            }
          }
          virtual={undefined}
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "alignSelf": null,
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 15,
                    },
                    Object {
                      "color": "#999999",
                      "fontSize": 13,
                      "marginLeft": 0,
                      "textAlign": "right",
                      "textDecorationLine": "underline",
                    },
                  ]
                }
                uppercase={false}
                virtual={undefined}
              >
                この続きを読む
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
